### PR TITLE
Artifactory Release Lifecycle Management - Support release bundle creation by aql and artifacts, support file specs

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-cli/CONTRIBUTING.md#tests) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.

---
Depends on https://github.com/jfrog/jfrog-cli-core/pull/1142

### CODE REVIEW

File: lifecycle/cli.go [hunk 1]
```diff
@@ -75,10 +77,19 @@ func validateCreateReleaseBundleContext(c *cli.Context) error {
...
-		return errorutils.CheckErrorf("exactly one of the following options must be supplied: --%s or --%s", cliutils.Builds, cliutils.ReleaseBundles)
+	return assertValidCreationMethod(c)
+}
+
+func assertValidCreationMethod(c *cli.Context) error {
+	methods := []bool{
+		c.IsSet("spec"), c.IsSet(cliutils.Builds), c.IsSet(cliutils.ReleaseBundles)}
+	if coreutils.SumTrueValues(methods) > 1 {
+		return errorutils.CheckErrorf("exactly one creation source must be supplied: --%s, --%s or --%s.\n"+
+			"The spec option is the recommended approach.", "spec", cliutils.Builds, cliutils.ReleaseBundles)
```
Comment (yahavi) [inline lifecycle/cli.go:88]: If `--builds` and `--release-bundles` are deprecated, I suggest rephrasing "The spec option is the recommended approach" to something like "The --builds and --release-bundles options are deprecated."

File: lifecycle/cli.go [hunk 2]
```diff
@@ -88,15 +99,23 @@ func create(c *cli.Context) (err error) {
 		return err
 	}
 
+	var creationSpec *spec.SpecFiles
+	if c.IsSet("spec") {
+		creationSpec, err = cliutils.GetSpec(c, true)
+		if err != nil {
+			return
+		}
+	}
```
Comment (yahavi) [inline lifecycle/cli.go:108]: Asking for an input pattern via an argument (`jf rtc <release bundle name> <release bundle version> <patren>`) will likely be much simpler and provide a better user experience. I think it'll be the most common scenario.
```suggestion
	if c.IsSet("spec") {
		creationSpec, err = cliutils.GetSpec(c, true)
	} else {
		creationSpec = createDefaultReleaseBundleSpec(c) // <- implement me
	}
```

File: lifecycle_test.go [hunk 3]
```diff
@@ -52,17 +53,84 @@ func TestLifecycle(t *testing.T) {
...
+}
+
+func TestReleaseBundleCreationFromArtifacts(t *testing.T) {
+	testReleaseBundleCreation(t, tests.UploadDevSpec, tests.LifecycleArtifacts, tests.GetExpectedLifecycleCreationByArtifacts())
+}
+
+func testReleaseBundleCreation(t *testing.T, uploadSpec, creationSpec string, expected []string) {
+	cleanHomeDir := initLifecycleTest(t)
+	defer cleanHomeDir()
+	defer cleanLifecycleTests(t)
```
Comment (yahavi) [inline lifecycle_test.go:99]: Can we put the cleanLifecycleTests in the callback of initLifecycleTest?

**Review by yahavi — CHANGES_REQUESTED** [review]

File: lifecycle/cli.go (hunk 2 above)
Comment (RobiNino) [reply lifecycle/cli.go:108]: We can consider adding this support upon requests on a later stage

File: lifecycle/cli.go [hunk 4]
```diff
@@ -75,10 +77,19 @@ func validateCreateReleaseBundleContext(c *cli.Context) error {
...
+		c.IsSet("spec"), c.IsSet(cliutils.Builds), c.IsSet(cliutils.ReleaseBundles)}
+	if coreutils.SumTrueValues(methods) > 1 {
+		return errorutils.CheckErrorf("exactly one creation source must be supplied: --%s, --%s or --%s.\n"+
+			"The spec option is the recommended approach.", "spec", cliutils.Builds, cliutils.ReleaseBundles)
+	}
+	// If the user did not provide a source, we suggest only the recommended spec approach.
+	if coreutils.SumTrueValues(methods) == 0 {
+		return errorutils.CheckErrorf("the --spec option is mandatory")
 	}
 	return nil
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply lifecycle/cli.go:94]: We decided to hide them at the first stage, and consider adding a warning on a later stage

**Pull request merged**